### PR TITLE
VG-10 Mission page

### PR DESCRIPTION
**Pull request by Arthaclarius**
## Tickets

https://x-team-internal.atlassian.net/browse/VG-10

## Demo

[image](https://vimeo.com/777364079/219f661a31)

### DB Changes

- Added release date to the Mission table
- Removed the finished column from the Game table. It was unnecessary.
- Updated the Seed function to add 3 missions

### Error handling

Added a folder with error classes to have more consistency while throwing errors. Updated the resolvers to use these new classes.

### Date Scalar

Added a scalar to handle dates

### `startGame` Resolver

When starting a game, it will check if there is already a game running. If so, it throws an error.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -45,11 +45,12 @@ model SocialLogin {
 }
 
 model Mission {
-  id          Int    @id @default(autoincrement())
+  id          Int       @id @default(autoincrement())
   title       String
   description String
   type        String
   level       Int
+  releaseDate DateTime?
```
Comment (lucasiervolino) [inline prisma/schema.prisma:53]: Is it intended that `releaseDate` has an optional value? Just to make sure.

Comment (smaryka) [reply prisma/schema.prisma:53]: It might make sense for this to be required, and compare to current time to determine which missions are released.

**Review by smaryka — CHANGES_REQUESTED** [review]
Just need to resolve the use of `releaseDate`

**Comment by Arthaclarius** [comment]
> 

Little confused there. I left that column as optional in case a mission does not have a planned release date. Eventually, a release date can be an incoming date in case we want to release it on a specific date in the future. Could you provide me an example of a Mission that is not released?

**Comment by Arthaclarius** [comment]
> 



> > It might make sense for this to be required, and compare to current time to determine which missions are released.
> 
> Little confused there. I left that column as optional in case a mission does not have a planned release date. Eventually, a release date can be an incoming date in case we want to release it on a specific date in the future. Could you provide me an example of a Mission that is not released?
> 
> @smaryka @lucasiervolino

For example, Mission 1 is already released and is available to the public, but missions 2 and 3 are not ready yet and do not have a planned date for their release, probably until the missions are coded and tested.

**Review by smaryka — APPROVED** [review]
I am fine with leaving the release date optional, and keep our options open. I'm not sure we will want to present unreleased missions to the user anyways, but that is a decision that can be taken later,

**Pull request merged**